docs/proposals/network-policy.md

```diff
@@ -0,0 +1,335 @@
+	// If this field is present and contains at least on item, this rule allows traffic only if the 
+	// traffic matches at least one item in the from list. 
+	From *[]NetworkPolicyPeer `json:"from,omitempty"`
+}
+
+type NetworkPolicyPort struct {
+	// Optional.  The protocol (TCP or UDP) which traffic must match.
+	// If not specified, this field defaults to TCP. 
+	Protocol *api.Protocol `json:"protocol,omitempty"`
+	
+	// If specified, the port on the given protocol.  This can 
+	// either be a numerical or named port.  If this field is not provided,
+	// this matches all port names and numbers.
+	// If present, only traffic on the specified protocol AND port
+	// will be matched.
+	Port *intstr.IntOrString `json:"port,omitempty"`
+}
+
+type NetworkPolicyPeer struct {
+	// Exactly one of the following must be specified.
+
+	// This is a label selector which selects Pods in this namespace.
+	// This field follows standard unversioned.LabelSelector semantics.
+	// If not provided, this selector selects no pods.
+	// If present but empty, this selector selects all pods in this namespace.
+	PodSelector *unversioned.LabelSelector `json:"podSelector,omitempty"`
+	
+	// Selects Namespaces using cluster scoped-labels.  This 
+	// matches all pods in all namespaces selected by this label selector. 
+	// This field follows standard unversioned.LabelSelector semantics.
+	// If omited, this selector selects no namespaces.
+	// If present but empty, this selector selects all namespaces.
+	NamespaceSelector *unversioned.LabelSelector `json:"namespaceSelector,omitempty"`
+}
+```
+
+### Behavior
+
+The following pseudo-code attempts to define when traffic is allowed to a given pod when using this API.
+
+```python
+def is_traffic_allowed(traffic, pod):
+  """
+  Returns True if traffic is allowed to this pod, False otherwise.
+  """
+  if not pod.Namespace.Spec.NetworkPolicy.Ingress.Isolation:
+    # If ingress isolation is disabled on the Namespace, all traffic is allowed.
+    return True 
+  elif traffic.source == pod.node.kubelet:
+    # Traffic is from kubelet health checks.
+    return True
+  else:
+    # If namespace ingress isolation is enabled, only allow traffic 
+    # that matches a network policy which selects this pod.
+    for network_policy in network_policies(pod.Namespace):
+      if not network_policy.Spec.PodSelector.selects(pod):
+        # This policy doesn't select this pod. Try the next one. 
+        continue
+
+      # This policy selects this pod.  Check each ingress rule 
+      # defined on this policy to see if it allows the traffic.
+      # If at least one does, then the traffic is allowed.
+      for ingress_rule in network_policy.Ingress or []:
+        if ingress_rule.matches(traffic): 
+          return True 
+
+  # Ingress isolation is DefaultDeny and no policies match the given pod and traffic.
+  # Use the cluster default behavior.
+  return clusterDefault(traffic, pod) 
```

Comment:
No, the cluster default is what applies when namespace.networkPolicy.ingress.isolation is undefined; otherwise traffic is refused if it is neither for probing (health checking or lifecycle) or allowed by a NetworkPolicy.
If this is not clear yet, just consider this: where does this code ever return "False"?